src/main.rs

```diff
@@ -719,6 +816,43 @@ fn run_cli(cli: Cli) -> Result<()> {
             }
             row.commit()?;
         }
+        Command::Path {
+            citation_key,
+            mkdir,
+        } => {
+            // Initialize the file directory path
+            let file_dir = if let Some(file_dir) = cli.attach_dir {
+                // at a user-provided path
+                info!(
+                    "Using user-provided file directory '{}'",
+                    file_dir.display()
+                );
+                file_dir
+            } else {
+                // at the default path
+                let default_db_path = data_dir.join("attachments");
+                info!(
+                    "Using default file directory '{}'",
+                    default_db_path.display()
+                );
+
+                default_db_path
+            };
+
+            // Extend with the filename.
+            let (record, row) =
+                get_record_row(&mut record_db, citation_key, &client, &config.on_insert)?
+                    .exists_or_commit_null("Cannot show directory for")?;
+            let Record { canonical, .. } = record;
+            let mut path = file_dir;
+            canonical.path_hash(&mut path);
+            if mkdir {
+                info!("Creating directory for canonical id '{canonical}'");
+                create_dir_all(&path)?;
+            }
+            println!("{}", path.display());
+            row.commit()?;
+        }
```

Comment:
This looks identical to the first few lines in the `Command::Attach` branch. Can we extract this?